File: src/AspNetToDoApi/AspNetToDoApi/Models/ToDoItem.cs
using Newtonsoft.Json;
using System;

namespace AspNetToDoApi.Models
{
    public class ToDoItem
    {
        public Guid Id { get; set; }

        public string Text { get; set; }

        public string Address { get; set; }

        public bool Done { get; set; }

        public ToDoItem()
        {
        }

        public ToDoItem(string name, string address, bool done)
        {
            Id = Guid.NewGuid();
            Text = name;
            Address = address;
            Done = done;
        }

        public void UpdateToDo(ToDoItem item)
        {
            Id = item.Id;
            Text = item.Text;
            Address = item.Address;
            Done = item.Done;
        }
    }
}

File: src/JavaToDoApi/src/main/java/com/fils/models/ToDoItem.java
package com.fils.models;

import java.util.UUID;

/**
 * Created by radu on 16/11/2016.
 */
public class ToDoItem {
    private UUID id;
    private String text;
    private String address;
    private boolean done;

    public ToDoItem(){

    }

    public ToDoItem(String text, String address, boolean done){
        this.id = UUID.randomUUID();
        this.text = text;
        this.address = address;
        this.done = done;
    }

    public void UpdateToDo(ToDoItem item){
        this.setId(item.getId());
        this.setText(item.getText());
        this.setAddress(item.getAddress());
        this.setDone(item.isDone());
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public boolean isDone() {
        return done;
    }

    public void setDone(boolean done) {
        this.done = done;
    }
}
